File: Game Client/game/WaitScreen.py
from .config import *
from .Button import *

class WaitScreen:
	def __init__(self, id, nb, nb_players, mode):
		self.mode = mode
		self.id = id
		self.nb_max = nb_players
		self.nb = nb
		self.font = pg.font.Font(font, int(winHeight * 0.085))
		self.size = [winWidth * 0.2, winHeight * 0.1]
		self.player_input = [Button("BACK TO MENU", (winWidth / 2) - (self.size[0] / 2), winHeight - self.size[1] - 25 , self.size[0], self.size[1], winHeight * 0.085)]


	def draw(self, win):
		for button in self.player_input:
			button.draw(win)
		text = self.font.render("WAITING FOR PLAYERS", True, (255, 255, 255))
		win.blit(text, ((winWidth / 2) - (text.get_size()[0] / 2), 25))
		text = self.font.render("ID : " + str(self.id), True, (255, 255, 255))
		win.blit(text, (10, 25))
		text = self.font.render(str(self.nb) + " / " + str(self.nb_max), True, (255, 255, 255))
		win.blit(text, ((winWidth / 2) - (text.get_size()[0] / 2), 60 + text.get_size()[1]))

	async def click(self, core, mousePos):
		if self.player_input[0].hitbox.collidepoint(mousePos):
			core.state = "menu"
			core.mode = "none"
			if core.GameRoom:
				await core.GameRoom.send(json.dumps({'type' : 'quitGame', 'id' : core.id, 'cmd' : 'quitWait'}))
			else:
				await core.GameHub.send(json.dumps({'type' : 'quitGame', 'id' : core.id, 'cmd' : 'quitWait'}))

File: srcs/django_code/transcend/myapp/static/js/game/WaitScreen.js
import { Button } from "./Button.js";
import { canvas, ctx } from "./canvas.js";
import { is_colliding } from "./Hitbox.js";
import { Vec2 } from "./Vec2.js";

export class WaitScreen {
	constructor(id, nb, nb_players, mode) {
		this.mode = mode;
		this.id = id;
		this.nb = nb;
		this.nb_max = nb_players;
		this.size = [canvas.width * 0.2, canvas.height * 0.1];
		this.button = new Button("BACK TO MENU", canvas.width / 2 - this.size[0] / 2, canvas.height - this.size[1] * 1.5, this.size[0], this.size[1]);
	}

	draw() {
		this.button.draw();
		ctx.fillText("WAITING FOR PLAYERS", canvas.width / 2, this.size[1]);
		ctx.fillText("ID : " + this.id, canvas.width * 0.06, this.size[1]);
		ctx.fillText(this.nb + " / " + this.nb_max, canvas.width / 2, this.size[1] * 2.5);
	}

	click(core, mousePos) {
		const pos = new Vec2(mousePos[0], mousePos[1]);
		if (is_colliding(pos, [0, 0], this.button.hitbox.pos, this.size)) {
			core.state = "menu";
			core.mode = "none"
			if (core.GameRoom)
				core.GameRoom.send(JSON.stringify({'type' : 'quitGame', 'id' : core.id, 'cmd' : 'quitWait'}));
			else
				core.GameHub.send(JSON.stringify({'type' : 'quitGame', 'id' : core.id, 'cmd' : 'quitWait'}));
		}
	}
}